src/wallets/IdentityWallet.ts

```diff
@@ -61,33 +73,71 @@ export class IdentityWallet implements TLWallet {
     progressCallback?: any
   ): Promise<UserObject> {
     this.wallet = ethers.Wallet.createRandom()
+    this.identityAddress = calculateIdentityAddress(
+      this.identityFactoryAddress,
+      this.wallet.address
+    )
 
     const encryptedKeystore = await this.wallet.encrypt(
       password,
       typeof progressCallback === 'function' && progressCallback
     )
 
-    const deployIdentityEndpoint = 'identities'
+    const serializedWallet: string = this.serializeWallet(
+      encryptedKeystore,
+      this.identityAddress
+    )
+
+    return {
+      address: this.address,
+      pubKey: 'Not implemented yet',
+      serializedWallet
+    }
+  }
 
-    const identity = await this.provider.postToEndpoint<DeployIdentityResponse>(
+  /**
+   * Deploys a new identity contract on the chain
+   */
+  public async deployIdentity(): Promise<string> {
+    const messageHash: string = ethers.utils.solidityKeccak256(
+      ['bytes1', 'bytes1', 'address', 'address'],
+      [
+        '0x19',
+        '0x00',
+        this.identityFactoryAddress,
+        this.identityImplementationAddress
+      ]
+    )
+    const signature = await this.rawSignHash(messageHash)
+
+    const deployIdentityEndpoint = 'identities'
+    const response = await this.provider.postToEndpoint<DeployIdentityResponse>(
       deployIdentityEndpoint,
       {
-        ownerAddress: this.wallet.address
+        implementationAddress: this.identityImplementationAddress,
+        factoryAddress: this.identityFactoryAddress,
+        signature
       }
     )
+    if (this.address !== response.identity) {
+      throw new Error(
+        `Delegate did not deploy the right identity contract. Deployed ${
+          response.identity
+        } instead of ${this.address}`
+      )
+    }
 
-    this.identityAddress = identity.identity
+    return this.address
+  }
 
-    const serializedWallet: string = this.serializeWallet(
-      encryptedKeystore,
-      identity.identity
+  public async isIdentityDeployed(): Promise<boolean> {
+    const response = await this.provider.fetchEndpoint<any>(
+      `/identities/${this.address}`
     )
-
-    return {
-      address: identity.identity,
-      pubKey: 'Not implemented yet',
-      serializedWallet
+    if ('not found' in response) {
```

Comment:
This will not work as this.provider.fetchEndpoint will already throw an error. 
You will have to use try catch